relay-metrics/src/aggregation.rs

```diff
@@ -1824,49 +1913,46 @@ impl fmt::Debug for Aggregator {
     }
 }
 
-impl Actor for Aggregator {
-    type Context = Context<Self>;
-
-    fn started(&mut self, ctx: &mut Self::Context) {
-        relay_log::info!("aggregator started");
-
-        // Subscribe to shutdown
-        Controller::subscribe(ctx.address());
+impl Service for AggregatorService {
+    type Interface = Aggregator;
 
-        // TODO: Consider a better approach than busy polling
-        ctx.run_interval(FLUSH_INTERVAL, |slf, _context| {
-            slf.try_flush();
+    fn spawn_handler(mut self, mut rx: relay_system::Receiver<Self::Interface>) {
+        let (ref flush_tx, _) = self.flush_channel;
+        let sender = flush_tx.clone();
+        tokio::spawn(async move {
+            loop {
+                sender.send(Flush).await.ok();
+                // Sleep for duration definited in `FLUSH_INTERVAL` and trigger the flush again.
+                tokio::time::sleep(FLUSH_INTERVAL).await;
+            }
+        });
+        tokio::spawn(async move {
+            // XXX: this requires the old system and runtime, and cannot be started in new tokio
+            // runtime ???
+            // let mut shutdown = Controller::subscribe_v2().await;
```

Comment:
not sure i follow this?  healthcheck is the new runtime and uses this.  i just tried out this code and it seems to work?

Reply:
The code itself works as expected, but test does not work when I have to start 2 different services/actors in 2 different systems, and I get the following error:
```
thread 'tokio-runtime-worker' panicked at 'System is not running', /.cargo/registry/src/github.com-1ecc6299db9ec823/actix-0.7.9/src/system.rs:116:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'aggregation::tests::test_flush_bucket' panicked at 'called `Result::unwrap()` on an `Err` value: SendError', relay-metrics/src/aggregation.rs:2801:79
test aggregation::tests::test_flush_bucket ... FAILED

```